extendeddatacrossword.go

```diff
@@ -83,27 +84,37 @@ func (eds *ExtendedDataSquare) solveCrossword(
 		// Track if a single iteration of this loop made progress
 		progressMade := false
 
+		var mut sync.Mutex
+		var wg sync.WaitGroup
+		wg.Add(int(eds.width))
+
 		// Loop through every row and column, attempt to rebuild each row or column if incomplete
 		for i := 0; i < int(eds.width); i++ {
 			i := i
 
 			errs.Go(func() error {
+				defer wg.Done()
 				solvedRow, progressMadeRow, err := eds.solveCrosswordRow(i, rowRoots, colRoots)
 				if err != nil {
 					return err
 				}
 
+				mut.Lock()
+				defer mut.Unlock()
 				solved = solved && solvedRow
 				progressMade = progressMade || progressMadeRow
 				return nil
 			})
 
 			errs.Go(func() error {
+				wg.Wait()
```

Comment:
Instead of adding wait groups and subroutines for inactive tasks would it be simpler if we just had two for loops, one for rows and one for columns?